src/dmd/frontend.d

```diff
@@ -79,11 +79,11 @@ string findDMDConfig(const(char)[] dmdFilePath)
     import dmd.dinifile : findConfFile;
     import std.string : fromStringz, toStringz;
 
-    auto f = findConfFile(dmdFilePath.toStringz, "dmd.conf");
+    auto f = findConfFile(dmdFilePath, "dmd.conf");
     if (f is null)
         return null;
 
-    return f.fromStringz.idup;
+    return f.idup;
```

Comment:
You can `idup` a string that is `null`, which will save you the if statement.

Reply:
```diff
diff --git a/src/dmd/frontend.d b/src/dmd/frontend.d
index 93c5a8a94..9ae64a51a 100644
--- a/src/dmd/frontend.d
+++ b/src/dmd/frontend.d
@@ -79,11 +79,7 @@ string findDMDConfig(const(char)[] dmdFilePath)
     import dmd.dinifile : findConfFile;
     import std.string : fromStringz, toStringz;

-    auto f = findConfFile(dmdFilePath, "dmd.conf");
-    if (f is null)
-        return null;
-
-    return f.idup;
+    return findConfFile(dmdFilePath, "dmd.conf").dup;
 }

 /**
```

Updated thanks